stix2validator/test/multi_obj_test.py

```diff
@@ -0,0 +1,50 @@
+import copy
+import json
+from . import ValidatorTest
+from .. import validate_string
+
+
+MULTI_OBJ_JSON = """
+[
+    {
+      "type": "attack-pattern",
+      "id": "attack-pattern--0c7b5b88-8ff7-4a4d-aa9d-feb398cd0061",
+      "created": "2016-02-12T08:17:27.000Z",
+      "modified": "2016-02-12T08:17:27.000Z",
+      "name": "Spear Phishing",
+      "description": "...",
+      "external_references": [
+        {
+          "source_name": "capec",
+          "external_id": "CAPEC-463"
+        }
+      ]
+    },
+
+    {
+      "type": "identity",
+      "id": "identity--2d1c6ab3-5e4e-48ac-a32b-f0c01c2836a8",
+      "created": "2014-08-08T15:50:10.983Z",
+      "modified": "2014-08-08T15:50:10.983Z",
+      "name": "ACME Widget, Inc.",
+      "identity_class": "organization"
+    }
+]
+"""
+
+class MultiObjTestCases(ValidatorTest):
+    valid_objs = json.loads(MULTI_OBJ_JSON)
+
+    def test_both_valid(self):
+        obj_results = validate_string(MULTI_OBJ_JSON)
+        self.assertEqual(len(obj_results), 2)
+        self.assertTrue(all(r.is_valid for r in obj_results))
+
+    def test_one_valid_one_invalid(self):
+        objs = copy.deepcopy(self.valid_objs)
+        objs[1]["id"] = objs[1]["id"].replace("identity", "abc")
+        obj_results = validate_string(json.dumps(objs))
```

Comment:
I didn't like having to `json.dumps` back to a string again here.  The API is inconsistent: we have functions (in validator.py) to validate files, strings, text streams (something I added), and parsed JSON.  The first three have the same behavior, catching a `SchemaInvalidError` and appending an error result; the last one doesn't.  It will raise the exception.  I didn't want all the try-catch junk in the test case, so I converted back to a string so I could call a function which deals with the exception.  I think we need consistent behavior across all of these validation functions.  It would be nice if I could call a validation function with `objs` directly.

Reply:
I agree. Perhaps `validate_instance` is misnamed. It could be renamed (not sure what the new name would be), and a new `validate_instance` added that's a wrapper around your new `validate` function. What do you think?

Reply:
:+1: for consistency. The new `validate_instance` (or whatever we call it) should be able to validate a dictionary, dictionary-like thing (like a python-stix2 object), or list of either.

Reply:
@gtback That would be ideal, however I believe we will run into problems validating python-stix2 objects because it stores timestamps as datetime objects instead of as strings. This was one of the problems that caused me to pull the validator out of python-stix, at least for now.

`validate_string(json.dumps(objs))` avoids this problem, but seems redundant. It feels overcomplicated for the validator to have to know which properties are timestamps and convert them to strings, though.

Reply:
Oh yeah, I forgot. Why is it always timestamps!? 😱